src/detect-engine-uint.c

```diff
@@ -218,13 +218,165 @@ PrefilterPacketU32Compare(PrefilterPacketHeaderValue v, void *smctx)
     return false;
 }
 
-static bool g_detect_u32_registered = false;
+static bool g_detect_uint_registered = false;
 
-void DetectU32Register(void)
+void DetectUintRegister(void)
 {
-    if (g_detect_u32_registered == false) {
+    if (g_detect_uint_registered == false) {
         // register only once
         DetectSetupParseRegexes(PARSE_REGEX, &uint_pcre);
-        g_detect_u32_registered = true;
+        g_detect_uint_registered = true;
     }
 }
+
+//same as u32 but with u8
+int DetectU8Match(const uint8_t parg, const DetectU8Data *du8)
+{
+    switch (du8->mode) {
+        case DETECT_UINT_EQ:
+            if (parg == du8->arg1) {
+                return 1;
+            }
+            return 0;
+        case DETECT_UINT_LT:
+            if (parg < du8->arg1) {
+                return 1;
+            }
+            return 0;
+        case DETECT_UINT_GT:
+            if (parg > du8->arg1) {
+                return 1;
+            }
+            return 0;
+        case DETECT_UINT_RA:
+            if (parg > du8->arg1 && parg < du8->arg2) {
+                return 1;
+            }
+            return 0;
+        default:
+            BUG_ON("unknown mode");
+    }
+    return 0;
+}
+
+/**
+ * \brief This function is used to parse u8 options passed via some u8 keyword
+ *
+ * \param u8str Pointer to the user provided u8 options
+ *
+ * \retval DetectU8Data pointer to DetectU8Data on success
+ * \retval NULL on failure
+ */
+
+DetectU8Data *DetectU8Parse (const char *u8str)
+{
+    DetectU8Data u8da;
+    DetectU8Data *u8d = NULL;
+    char arg1[16] = "";
+    char arg2[16] = "";
+    char arg3[16] = "";
+
+#define MAX_SUBSTRINGS 30
```

Comment:
This is already defined in `detect-parse.h`

Reply:
Got it.